discovery-provider/src/api/v1/helpers.py

```diff
@@ -141,6 +141,7 @@ def extend_track(track):
         # NOTE: Legacy track segments store the duration as a string
         duration += float(segment["duration"])
     track["duration"] = round(duration)
+    track["downloadable"] = track["download"] and track["download"]["is_downloadable"]
```

Comment:
Does this work? If we need to check for the existence of download before indexing into it, but download isn't there, this indexing op to check for download will throw

Reply:
yeah i think you're right, i was in js head space. should be checking "download" in track as well